crates/valence_world_border/src/lib.rs

```diff
@@ -94,316 +34,215 @@ pub const DEFAULT_DIAMETER: f64 = (DEFAULT_PORTAL_LIMIT * 2) as f64;
-        }
-    }
+#[derive(Component, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub struct WorldBorderPortalTpBoundary(pub i32);
 
-    pub fn current_duration(&self) -> i64 {
-        let speed = self.duration - self.timestamp.elapsed().as_millis() as i64;
-        speed.max(0)
+impl Default for WorldBorderPortalTpBoundary {
+    fn default() -> Self {
+        Self(DEFAULT_PORTAL_LIMIT)
     }
 }
 
-/// An event for controlling world border diameter.
-/// Setting duration to 0 will move the border to `new_diameter` immediately,
-/// otherwise it will interpolate to `new_diameter` over `duration` time.
-/// ```
-/// # use bevy_ecs::event::EventWriter;
-/// # use valence_world_border::SetWorldBorderSizeEvent;
-/// # use core::time::Duration;
-/// fn change_diameter(
-///     mut event_writer: EventWriter<SetWorldBorderSizeEvent>,
-///     instance: bevy_ecs::entity::Entity,
-///     diameter: f64,
-///     duration: Duration,
-/// ) {
-///     event_writer.send(SetWorldBorderSizeEvent {
-///         instance,
-///         new_diameter: diameter,
-///         duration,
-///     })
-/// }
-/// ```
-#[derive(Event, Clone, Debug)]
-pub struct SetWorldBorderSizeEvent {
-    /// The instance to change border size. Note that this instance must contain
-    /// the [`WorldBorderBundle`] bundle
-    pub instance: Entity,
-    /// The new diameter of the world border
-    pub new_diameter: f64,
-    /// How long the border takes to reach it new_diameter in millisecond. Set
-    /// to 0 to move immediately.
-    pub duration: Duration,
-}
-
-fn wb_size_change(
-    mut events: EventReader<SetWorldBorderSizeEvent>,
-    mut instances: Query<(&WorldBorderDiameter, Option<&mut MovingWorldBorder>)>,
-) {
-    for SetWorldBorderSizeEvent {
-        instance,
-        new_diameter,
-        duration,
-    } in events.iter()
-    {
-        let Ok((diameter, mwb_opt)) = instances.get_mut(*instance) else {
-            continue;
-        };
-
-        if let Some(mut mvb) = mwb_opt {
-            mvb.new_diameter = *new_diameter;
-            mvb.old_diameter = diameter.get();
-            mvb.duration = duration.as_millis() as i64;
-            mvb.timestamp = Instant::now();
+impl Default for WorldBorderLerp {
+    fn default() -> Self {
+        Self {
+            current_diameter: DEFAULT_DIAMETER,
+            target_diameter: DEFAULT_DIAMETER,
+            remaining_ticks: 0,
         }
     }
 }
 
-fn border_for_player(
-    mut clients: Query<(&mut Client, &Location), Changed<Location>>,
-    wbs: Query<
-        (
-            &WorldBorderCenter,
-            &WorldBorderWarnTime,
-            &WorldBorderWarnBlocks,
-            &WorldBorderDiameter,
-            &WorldBorderPortalTpBoundary,
-            Option<&MovingWorldBorder>,
-        ),
-        With<Instance>,
-    >,
+fn init_world_border_for_new_clients(
+    mut clients: Query<(&mut Client, &VisibleChunkLayer), Changed<VisibleChunkLayer>>,
```

Comment:
does this make world borders act per chunk layer?

Reply:
Yep. I'll explain that in the docs. I didn't think it made much sense to act on entity layers since clients can view >1 entity layers.